src/function.rs

```diff
@@ -72,8 +72,8 @@ impl<'a> FnAttribute<'a> {
         }
     }
 
-    fn to_sys(&self) -> gccjit_sys::gcc_jit_fn_attribute {
-        match *self {
+    fn to_sys(self) -> gccjit_sys::gcc_jit_fn_attribute {
+        match self {
```

Comment:
Please keep the original version.
I'm assuming you'll have to remove the `derive(Copy)` to make clippy happy, but it's good to remove it since this type is 24 bytes.